### PR TITLE
[SPARK-20541][SPARKR][SS] support awaitTermination without timeout

### PR DESCRIPTION
**Pull request by felixcheung**
## What changes were proposed in this pull request?

Add without param for timeout - will need this to submit a job that runs until stopped
Need this for 2.2

## How was this patch tested?

manually, unit test

### CODE REVIEW

File: R/pkg/inst/tests/testthat/test_streaming.R [hunk 1]
```diff
@@ -61,6 +61,7 @@ test_that("read.stream, write.stream, awaitTermination, stopQuery", {
 
   stopQuery(q)
   expect_true(awaitTermination(q, 1))
+  expect_error(awaitTermination(q), NA)
```
Comment (shivaram) [inline R/pkg/inst/tests/testthat/test_streaming.R]: Why will this generate an error ?

**Comment by shivaram** [comment]
LGTM. Minor comment

File: R/pkg/inst/tests/testthat/test_streaming.R (hunk 1 above)
Comment (felixcheung) [inline R/pkg/inst/tests/testthat/test_streaming.R]: `NA` means expects no error ;)
yea, it's not very intuitive. There is no return value in this case, so just checking to make sure it doesn't throw an error

**Comment by felixcheung** [comment]
merged to master/2.2
thanks for the review!

**Pull request closed without merging**